testing/citest/tests/appengine_smoke_test.py

```diff
@@ -0,0 +1,314 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Integration test for App Engine. Primarily tests the deploy operation and upsert load balancer pipeline stage,
+which are relatively complex and not well covered by unit tests.
+
+Sample Usage:
+    Assuming you have created $PASSPHRASE_FILE (which you should chmod 400)
+    and $CITEST_ROOT points to the root directory of this repository
+    (which is . if you execute this from the root)
+
+  PYTHONPATH=$CITEST_ROOT \
+    python spinnaker/testing/citest/tests/appengine_smoke_test.py \
+    --gce_ssh_passphrase_file=$PASSPHRASE_FILE \
+    --gce_project=$PROJECT \
+    --gce_zone=$ZONE \
+    --gce_instance=$INSTANCE 
+or
+  PYTHONPATH=$CITEST_ROOT \
+    python spinnaker/testing/citest/tests/appengine_smoke_test.py \
+    --native_hostname=host-running-smoke-test
+"""
+
+import sys
+
+import citest.gcp_testing as gcp
+import citest.json_contract as jc
+import citest.json_predicate as jp
+import citest.service_testing as st
+
+import spinnaker_testing as sk
+import spinnaker_testing.gate as gate
+import spinnaker_testing.frigga as frigga
+import citest.base
+
+
+class AppengineSmokeTestScenario(sk.SpinnakerTestScenario):
+  """Defines the scenario for the integration test.
+  
+  We're going to:
+    Create a Spinnaker Application
+    Create a Spinnaker Server Group (implicitly creates a Spinnaker Load Balancer)
+    Create a Pipeline with the following stages
+      - Deploy
+      - Upsert Load Balancer
+    Delete Load Balancer (implicitly destroys the Spinnaker Server Groups created within this test)
+    Delete Application
+  """
+  @staticmethod
+  def new_agent(bindings):
+    return gate.new_agent(bindings)
+  
+  def __init__(self, bindings, agent=None):
+    super(AppengineSmokeTestScenario, self).__init__(bindings, agent)
+    
+    self.TEST_APP = bindings['TEST_APP']
+    self.TEST_STACK = bindings['TEST_STACK']
+    self.__path = 'applications/%s/tasks' % self.TEST_APP
+    self.__gcp_project = bindings['APPENGINE_PRIMARY_MANAGED_PROJECT_ID']
+    self.__cluster_name = frigga.Naming.cluster(self.TEST_APP, self.TEST_STACK)
+    self.__server_group_name = frigga.Naming.server_group(self.TEST_APP, self.TEST_STACK)
+    self.__lb_name = 'citest'
```

Comment:
If this is under your control, dont use a constant name. If it is derived, dont have it derive constant names. You want multiple tests to be able to run without necessarily conflicting with one another. If one test leaks the load balancer you dont want that to influence another run.

The google_smoke_test does this:
   self.__lb_name = '{app}-{stack}-{detail}'.format(
	app=bindings['TEST_APP'], stack=bindings['TEST_STACK'],
        detail=self.__lb_detail)

It should be updated to use frigga as you are elsewhere already.